rfcs/0015-pablo-internals-restructure.md

```diff
@@ -0,0 +1,321 @@
+# RFC-0015
+
+The [current implementation of `DualAssetConstantProduct`][dacp] is tightly
+coupled with the pablo pallet, depending on pablo's `Config` trait directly for
+all of it's functionality and directly modifying pablo's storage. This makes it
+very difficult to test, since a full runtime is required to provide the required
+mocks for it to work properly.
+
+## Goals
+
+1. Remove `DualAssetConstantProduct`'s direct dependency on pablo
```

Comment:
These are not goals, they are a solution to a problem. Problem should come first before the solution.
refer https://en.wikipedia.org/wiki/Circular_reasoning

Reply:
Is the opening statement insufficient as a description of the problem? I can elaborate if so